src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,125 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static final int CAPACITY = 10;
+    private T[] elementData;
+    private int size = 0;
+    private int currentCapacity;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[CAPACITY];
+        currentCapacity = CAPACITY;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == currentCapacity) {
+            elementData = ensureCapacity();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == currentCapacity) {
+            elementData = ensureCapacity();
+        }
+        if (index > size) {
+            throw new ArrayIndexOutOfBoundsException("Wrong index!");
+        }
+        if (index < size) {
+            System.arraycopy(elementData, index, elementData, index + 1, currentCapacity - size);
```

Comment:
```suggestion
            System.arraycopy(elementData, index, elementData, index + 1, ??? );
```
I think this part is not correct

Reply:
done